parallel_expr/parallel_expr.cpp

```diff
@@ -11,26 +12,24 @@
  * Both the group_size and num_groups are modified by these experiments
  */
 int main(int argc, char** argv) {
-  if (argc != 4) {
+  if (argc != 5) {
     std::cout << "Incorrect number of arguments. "
                  "Expected three but got " << argc-1 << std::endl;
-    std::cout << "Arguments are: input_stream, output_file, max_threads" << std::endl;
+    std::cout << "Arguments are: input_stream, output_file, csv_out_file, max_threads" << std::endl;
     exit(EXIT_FAILURE);
   }
 
   std::string input  = argv[1];
   std::string output = argv[2];
-  int max_thr = std::stoi(argv[3]);
+  std::string csv_file = argv[3];
+  int max_thr = std::stoi(argv[4]);
   if (max_thr <= 0 || max_thr > 100) {
     printf("Argument CPU cores invalid. Out of range [1,100");
     exit(EXIT_FAILURE);
   }
   // max_thr for group size experiment needs to be divisible by 20
   int gr_max_thr = max_thr - (max_thr % 20);
 
-  // backup the current configuration
-  backup_configuration();
-
   std::vector<int> sizes{1,2,4,10,20};
 
   // Group Size Experiment
```

Comment:
If we aren't going to make a CSV for this experiment and we only briefly mention the concept of this test in the paper... should we just get rid of it?

Reply:
iirc we use this to create fig 14

Reply:
We use the num_groups experiment to create fig 14. However, we don't do anything with the group_size experiment. This file runs both of these experiments.